pkg/planner/core/rule_predicate_simplification.go

```diff
@@ -147,6 +190,116 @@ func applyPredicateSimplification(sctx base.PlanContext, predicates []expression
 	return newValues
 }
 
+// Check for constant false condition.
+func unsatisfiableExpression(ctx base.PlanContext, p expression.Expression) bool {
+	if constExpr, ok := p.(*expression.Constant); ok {
+		if b, err := constExpr.Value.ToBool(ctx.GetSessionVars().StmtCtx.TypeCtx()); err == nil && b == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func unsatisfiable(ctx base.PlanContext, p1, p2 expression.Expression) bool {
+	var equalPred expression.Expression
+	var otherPred expression.Expression
+	col1, p1Type := findPredicateType(p1)
+	col2, p2Type := findPredicateType(p2)
+	if col1 != col2 || col1 == nil {
+		return false
+	}
+	if p1Type == equalPredicate {
+		equalPred = p1
+		otherPred = p2
+	} else if p2Type == equalPredicate {
+		equalPred = p2
+		otherPred = p1
+	}
+	if equalPred == nil || otherPred == nil {
+		return false
+	}
+	// Copy constant from equal predicate into other predicate.
```

Comment:
just a easy explanation for this
```
expr1: col=$1
expr2: myfunc(col, z)
===> myfunc($1,z)
```
then use PropagateConstant to converge new expr: like(a>1 and a<1) ===> false

Reply:
This case actually is not covered by PropagateConstant if the type of col and constant are not the same.

Reply:
i see.